### DISCUSSION TITLE
With the Matrox Mystique now un-dev-branched, has anybody managed to build a setup that can actually run the Matrox Mystique Gaming Bundle?

### DISCUSSION THREAD

**Discussion by tokariu** [86Box/86Box#4376] [Q&A]
Before I start, let me say a word to the devs involved with the Matrox/MGA emulation development: Thank You! 
I'm very grateful that this piece of hardware is being worked on and I really appreciate the progress that has been achieved in the past year! This is really awesome and one day it is going to fill one of the last major puzzle pieces of the early 3D Accelerator era.

That being said, I recently tried to change my old Win98SE setup that I have been using with a 86box instance to make use of the new Matrox Mystique card.

Unfortunately I failed building a setup that was able to run the games that originally shipped with the Matrox Mystique Card (I still own the original CDROMs, but I lost the GPU ages ago).
The Games were the Mystique Editions of Mechwarrior 2, Destruction Derby 2 and Scorched Planet.

The Problem is the same for all these games: after starting the game it seems to work for the first 1 or 2 seconds, the initial frames will be rendered and I see movement, but shortly after the whole 86box instance hangs and crashes.

At this point I don't know if the reason for the problem is the setup of my 86Box hardware, some software/OS/driver issue, or just the complexity of the MGA emulation that is just not working good enough to be able to play games yet.

Maybe someone can tell me more about the current status of the Mystique and this issue. I saw @Cacodemon345 was very active fixing things for Mystique recently.

Details for the Setup I used:

86Box and ROM set v4.1.1
Win98SE
PCI Matrox Mystique (non-220)
3dfx voodoo graphics unchecked
128MB RAM
[i430FX] ASUS P/I-P55TP4XE
Socket 7 (single voltage)
Intel Pentium OverDrive MMX 166MHz

I also tried another motherboard with the same results:
(same as above, but..)
[I430HX] ASUS P/I-P55T2P4 and/or
[I430VX] ASUS P/I-P55TVP4
with Intel Pentium MMX 166 MHz

Is there anything I can choose that works better with the Matrox Mystique card? Or doesn't it matter and it's just the mystique emulation that is not yet able to run these games?

**Comment by OBattler** [comment]
What driver are you using specifically for the Mystique?

**Comment by tokariu** [comment]
I got it working! I tinkered with so many settings and things in the meanwhile that I'm not 100% sure what the root cause was. But I have a clue..

I removed the mystique device in win98, uninstalled the drivers.
reinstalled the mystique drivers from the original Bundle CDROM. I noticed the MGA performance settings might be the reason for the crashes. I probably have changed settings there before without knowing. It seems the default settings seems to be working now.

but I also upgraded 86box version to 4.2, made sure I get 100% emulation speed (was too slow before), changed render output through all option until I stuck with vulcan, and many other little things. I also had an instance running that used the Matrox Mystique drivers that windows98SE installed without the matrox driver CDROM, then the MGA options are missing in the display options control panel and the Box is then also crashing when trying to run the games.

This are the drivers I reinstalled:
![Zwischenablagebild](https://github.com/86Box/86Box/assets/7195846/179f50c8-af48-472d-a31f-f3d6f59fcd8a)

and this are the default performance settings after the reinstall (and these actually do already work):
![Zwischenablagebild (1)](https://github.com/86Box/86Box/assets/7195846/d1561a39-0070-4853-b1d7-50f1888fe762)

i probably changed these settings to this, and then it is crashing:
![Zwischenablagebild (2)](https://github.com/86Box/86Box/assets/7195846/18562704-6aaf-4d89-9ad3-52344d392063)

I tried playing around more with these settings (each change requires a reboot..) and it turns out that when I turn on "use Bus-Mastering" setting, the games crashes everytime. 
I can activate "3D Acceleration" (3D-Beschleunigung in the screenshot) without problem, though. 

So I'm just guessing but it turns out to be the Bus-Mastering setting that caused the crashes.

**Comment by tokariu** [comment]
When I run DD2 I notice that the menu screen is having very visible "tiles". I don't know if that was the original behaviour. PS1, Windows and DOS versions of DD2 didn't have these visible tiles. Could this be an issue with the Matrox emulation or am I just not remembering original behaviour back in the 90s?
![Zwischenablagebild (4)](https://github.com/86Box/86Box/assets/7195846/c45ab12b-c9cf-462f-a70b-c934bb842e0f)

**Comment by tokariu** [comment]
just for fun I tried to change the Mystique with the Mystique 220 and then I'm back to the crashes as before again.

With the Mystique 220 and the drivers that come with win98se, all games crashes, even the demos on the Mystique 220 CD.
I also tried to install the original Mystique 220 Bundle drivers, and then I do get the extra MGA options in the display settings. However, there is no "Bus Mastering" option to disable anywhere available in the Mystique 220 MGA options. Maybe that's why it will always crashes on this card.

I also noticed that the CPU load while starting a game is unproportinal higher compared to the normal Mystique (i know the 220 has more MHz and obviously requires more power to emulate, but thats huge..) 
with the normal Mystique I get 100% emulation speed with my intel i7 core 10th gen cpu maxed out. with the Mystique 220 I can only get up to like 50-60% emulation speed anymore. and only for a short period until it crashes.

is the 220 version really so much more cpu demanding? and could the slow emulation speed also be the reason for the crashes?

EDIT:
and again.. it looks it was a driver problem again. I installed the latest legacy driver from matrox website, found the MGA settings in the systray and could disable bus mastering.

ALSO: with the windows driver for the Mystique'S i can barely emulate with 50-60% speed. After installing the correct and working bulk or latest legacy driver from matrox directly, it can emulate at 100% speed again.

**Comment by tokariu** [comment]
I just stumbled upon a main menu screenshot that was taken from the real matrox mystique hardware: (credits to anakin94):

![DD2MMenu](https://github.com/86Box/86Box/assets/7195846/35f941f6-a625-4a1c-ae2f-cc7463601db0)

this is how it looks on real hardware, there are no visible tiles.

![324207269-c45ab12b-c9cf-462f-a70b-c934bb842e0f](https://github.com/86Box/86Box/assets/7195846/c4f2a66a-4fc8-4b02-b539-4a077aab95b9)

whereas in 86Box matrox mystique the tiles are very much visible. I played a lot with matrox settings and drivers, even monitors, and the 220 model, but nothing removes the tiles. any ideas where these come from?
